.github/workflows/compose-rtdv3-verify.yaml

```diff
@@ -0,0 +1,212 @@
+        description: "The revision sha"
+        required: true
+        type: string
+      GERRIT_PROJECT:
+        description: "Project in Gerrit"
+        required: true
+        type: string
+      GERRIT_REFSPEC:
+        description: "Gerrit refspec of change"
+        required: true
+        type: string
+      TARGET_REPO:
+        # yamllint disable-line rule:line-length
+        description: "The target GitHub repository needing the required workflow"
+        required: true
+        type: string
+    secrets:
+      RTD_TOKEN:
+        description: "RTD API user token"
+        required: true
+
+concurrency:
+  # yamllint disable-line rule:line-length
+  group: compose-rtdv3-verify-${{ github.workflow }}-${{ github.event.inputs.GERRIT_CHANGE_ID || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  READTHEDOCS_FOUND: true
+  TOX_ENVS: docs,docs-linkcheck
+  TOX_DIR: docs/
+  DOC_DIR: _build/html
+  PARALLEL: true
+  DEFAULT_VERSION: latest
+  MASTER_RTD_PROJECT: doc
+
+jobs:
+  rtd-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Gerrit Checkout
+        # yamllint disable-line rule:line-length
+        uses: lfit/checkout-gerrit-change-action@c70c41f86f57a62ae26ac17b4f16516212c46222  # v0.5
+        with:
+          gerrit-refspec: ${{ inputs.GERRIT_REFSPEC }}
+          delay: "0s"
+          submodules: "true"
+      - uses: lfit/checkout-gerrit-change-action@v0.4
+        with:
+          gerrit-refspec: ${{ inputs.GERRIT_REFSPEC }}
+          delay: "0s"
+          repository: ${{ inputs.TARGET_REPO }}
+          ref: refs/heads/${{ inputs.GERRIT_BRANCH }}
+      - uses: actions/setup-python@v4
+        id: setup-python
+        with:
+          python-version: "3.11"
+      - name: Verify If readthedocs Config exists
+        run: |
+          echo "Verifying if readthedocs.yaml config file exists"
+          if [ ! -d .readthedocs.yaml ] ; then
+            echo "INFO Config file not found. Skipping further checks."
+            echo "READTHEDOCS_FOUND=false" >> "$GITHUB_ENV"
+          fi
+      - name: Installing dependencies
+        if: ${{ env.READTHEDOCS_FOUND == 'true' }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install lftools niet~=1.4.2 cryptography<3.4 yq tox
+          # urllib3 needs to be pinned to avoid timeouts
+          pip install --upgrade urllib3~=1.26.15
+          mkdir -p "${GITHUB_WORKSPACE}/.config/lftools/lftools.ini"
+          cat << EOF > "${GITHUB_WORKSPACE}/.config/lftools/lftools.ini"
+          [rtd]
+          endpoint=https://readthedocs.org/api/v3/
+          token=${{ secrets.RTD_TOKEN }}
+          EOF
+      - name: Running tox
+        # yamllint disable rule:line-length
+        if: ${{ env.READTHEDOCS_FOUND == 'true' }}
+        run: |
+          PATH="${GITHUB_WORKSPACE}/.local/bin:${PATH}"
+
+          ARCHIVE_TOX_DIR="${GITHUB_WORKSPACE}/archives/tox"
+          ARCHIVE_DOC_DIR="${GITHUB_WORKSPACE}/archives/docs"
```

Comment:
What is the point of all of this ARCHIVE_*_DIR stuff if there is nothing that saves the directories after the workflow completes. Please look into if there is a way to do with with GHA or strip out all this unneeded extra faffing around with archive dirs